File: python/data_cleaning.py
def handle_missing_values(data, strategy="mean", columns=None):
    """Handle missing values in the dataset."""
    if columns is None:
        columns = data.columns
    for column in columns:
        if strategy == "mean":
            data[column].fillna(data[column].mean(), inplace=True)
        elif strategy == "median":
            data[column].fillna(data[column].median(), inplace=True)
        elif strategy == "drop":
            data.dropna(subset=[column], inplace=True)
    return data

File: r/data_cleaning.R
handle_missing_values <- function(data, strategy = "mean", columns = NULL) {
  if (is.null(columns)) {
    columns <- colnames(data)
  }
  for (column in columns) {
    if (strategy == "mean") {
      data[[column]][is.na(data[[column]])] <- mean(data[[column]], na.rm = TRUE)
    } else if (strategy == "median") {
      data[[column]][is.na(data[[column]])] <- median(data[[column]], na.rm = TRUE)
    } else if (strategy == "drop") {
      data <- data[!is.na(data[[column]]), ]
    }
  }
  return(data)
}